sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala

```diff
@@ -208,6 +208,15 @@ class ExpressionParserSuite extends AnalysisTest {
     assertEqual("a rlike 'pattern\\t\\n'", 'a rlike "pattern\\t\\n", parser)
   }
 
+  test("(NOT) LIKE (ANY | ALL) expressions") {
+    assertEqual("a like any ('foo%', 'bar%')", ('a like "foo%") || ('a like "bar%"))
+    assertEqual("a not like any ('foo%', 'bar%')", !('a like "foo%") || !('a like "bar%"))
+    assertEqual("not (a like any ('foo%', 'bar%'))", !(('a like "foo%") || ('a like "bar%")))
+    assertEqual("a like all ('foo%', 'bar%')", ('a like "foo%") && ('a like "bar%"))
+    assertEqual("a not like all ('foo%', 'bar%')", !('a like "foo%") && !('a like "bar%"))
+    assertEqual("not (a like all ('foo%', 'bar%'))", !(('a like "foo%") && ('a like "bar%")))
```

Comment:
Could you add two more tests for error handling for L1396 and L1422 in AstBuilder.scala?